src/cpp/fastdds/subscriber/DataReader.cpp

```diff
@@ -55,6 +55,24 @@ DataReader::~DataReader()
 {
 }
 
+ReturnCode_t DataReader::enable()
+{
+    if (enable_)
+    {
+        return ReturnCode_t::RETCODE_OK;
+    }
+
+    if (false == impl_->get_subscriber()->is_enabled())
+    {
+        return ReturnCode_t::RETCODE_PRECONDITION_NOT_MET;
+    }
+
+    enable_ = true;
```

Comment:
This should not be necessary, we are setting the value after enabling the implementation.